File: client_android/app/src/main/java/com/securecall/app/ghostnet/media/MediaRouterInboundStub.java
package com.securecall.app.ghostnet.media;

import android.util.Log;

/**
 * MEDIA_ROUTER_INBOUND
 *
 * Minimal inbound media router:
 *  - receives decoded PCM frames,
 *  - forwards them to AudioPlaybackStub.
 */
public class MediaRouterInboundStub {

    private static final String TAG = "MEDIA_ROUTER_INBOUND";

    public static void handleDecodedPcm(byte[] pcm) {
        if (pcm == null) {
            return;
        }
        Log.d(TAG, "handleDecodedPcm(): got " + pcm.length + " bytes of PCM, forwarding to AudioPlaybackStub");
        AudioPlaybackStub.enqueuePcm(pcm);
    }
}

File: client_android/app/src/main/java/com/securecall/app/ghostnet/media/MediaRouterInboundStub.kt
package com.securecall.app.ghostnet.media

import android.util.Log

/**
 * Minimal inbound media router:
 * - receives decoded PCM frames,
 * - forwards them to AudioPlaybackStub.
 */
object MediaRouterInboundStub {

    private const val TAG = "MEDIA_ROUTER_INBOUND"

    @JvmStatic
    fun handleDecodedPcm(pcm: ByteArray) {
        Log.d(TAG, "handleDecodedPcm(): got \${pcm.size} bytes of PCM, forwarding to AudioPlaybackStub")
        AudioPlaybackStub.enqueuePcm(pcm)
    }
}
